src/service/filtering/generic_view.py

```diff
@@ -0,0 +1,39 @@
+from src.common.product_models.columnar_attribs_common_models import ColumnarAttributesSpec, AttributesColumnSpec
+
+_FIELD_COL = {
+    "kbase_display_name": {
+        "type": "string",
+        "filter_strategy": "ngram",
+    }
+}
+
+_GENERIC_VIEW_NAME_SUFFIX = "_generic_view"
+
+_GENERIC_VIEW_PRODUCTS = ['microtrait', 'biolog']
+
+
+def create_generic_spec() -> ColumnarAttributesSpec:
+    """
+    Create a generic column spec for a collection.
+
+    This spec only includes fields common to most or all collections that require special handling and it is
+    expected that all other fields are handled in a generic way.
+    """
+
+    columns = [AttributesColumnSpec(key=key, **specs) for key, specs in _FIELD_COL.items()]
+
+    return ColumnarAttributesSpec(columns=columns)
+
+
+def is_generic_view_product(data_product: str) -> bool:
```

Comment:
This can be checked by looking for the `requires_generic_view` flag on the data product spec - maybe add a method there to make it easy? Or a method to get the view name from the data product spec class

Reply:
E.g. https://github.com/kbase/collections/pull/582/files#diff-ab437ceb2ccb0562a34ddf1b5ecccc3db5b4382fc820c99c8c32b8d7439765f9R92

Reply:
Or instead of a flag you could maybe just hard code the view name. Haven't really thought that through though

Reply:
removed this method since there is no usage for this method now.

Reply:
:+1: